test/runner.spec.js

```diff
@@ -433,4 +433,87 @@ describe('Runner', function () {
       });
     });
   });
+
+  describe('.run', function () {
+    describe('"test end" event', function () {
+      it('should emit "test end" event for each test', function (done) {
+        var finishedTestTitles = [];
+
+        suite.addTest(new Test('first test', noop));
+        suite.addTest(new Test('second (failed) test', function () {
+          expect(1 + 1).to.equal(3);
```

Comment:
I'd prefer `expect(false).to.equal(true)` or better yet `throw new Error("FAIL TEST")` in place of all the `expect`s that are intended to fail in these new tests, for safety (no chance of changes to the variables and values involved ending up in a passing `expect` in the future), simplicity and clarity.